src/segment-loader.js

```diff
@@ -680,6 +682,22 @@ export default class SegmentLoader extends videojs.EventTarget {
       'fetchAtBuffer:', this.fetchAtBuffer_,
       'bufferedTime:', bufferedTime);
 
+    let syncPointError = 0;
+    if (currentTime < lastBufferedStart) {
```

Comment:
I think that handling the gap resolution logic may be better in playback watcher, probably within `fixedBadSeeks_`: https://github.com/videojs/videojs-contrib-hls/blob/2158b756eef8a2441fa580f673e316391d63ada6/src/playback-watcher.js#L157

This would also allow us to re-seek.